execution/binary/scalar.go

```diff
@@ -145,8 +147,11 @@ func (o *scalarOperator) loadSeries(ctx context.Context) error {
 	series := make([]labels.Labels, len(vectorSeries))
 	for i := range vectorSeries {
 		if vectorSeries[i] != nil {
-			lbls := labels.NewBuilder(vectorSeries[i]).Del(labels.MetricName).Labels(nil)
-			series[i] = lbls
+			lbls := labels.NewBuilder(vectorSeries[i])
```

Comment:
Maybe using `function.DropMetricName` instead of full `Builder` would save some mem, as we only drop name label here? We do the same in matrix selector. 🙂